core/src/main/java/org/apache/calcite/sql/dialect/HsqldbSqlDialect.java

```diff
@@ -35,7 +35,8 @@
 public class HsqldbSqlDialect extends SqlDialect {
   public static final SqlDialect DEFAULT =
       new HsqldbSqlDialect(EMPTY_CONTEXT
-          .withDatabaseProduct(DatabaseProduct.HSQLDB));
+          .withDatabaseProduct(DatabaseProduct.HSQLDB)
+          .withCaseSensitive(false));
```

Comment:
You can add test cases in `RelToSqlConverterTest`.

Reply:
Not sure that a test for SQL generation (RelToSqlConverterTest) will demonstrate a change to HSQLDB's validation behavior.

@lameyer I'm not sure that HSQLDB is case insensitive:
```
0: jdbc:hsqldb:res:scott> !connect jdbc:hsqldb:res:scott SCOTT TIGER
0: jdbc:hsqldb:res:scott> select count(*) from "EMP";
+-----------------+
|       C1        |
+-----------------+
| 14              |
+-----------------+
1 row selected (0.001 seconds)
0: jdbc:hsqldb:res:scott> select count(*) from "emp";
Error: user lacks privilege or object not found: emp (state=42501,code=-5501)
```

Reply:
Ah so I think it might only be case insensitive for unquoted column/table names, which is where I was running into this... Is there a way to set that in Calcite?